apps/mull-api/src/app/event/event.service.ts

```diff
@@ -47,33 +46,42 @@ export class EventService {
       .getMany();
   }
 
-  /**
-   * Find all the events that the user has not joined, created or is not co-hosting
-   * @param userId
-   */
-  findDiscoverEvent(userId: number): Promise<Event[]> {
-    return this.eventRepository
+  async isUserAttendingEvent(eventId: number, userId: number): Promise<boolean> {
+    const event = await this.eventRepository
       .createQueryBuilder('event')
+      .leftJoin('event.participants', 'user')
       .leftJoin('event.coHosts', 'coHost')
-      .leftJoin('event.participants', 'participant')
-      .leftJoin('event.host', 'host')
       .where(
         new Brackets((qb) => {
-          qb.where('coHost.id != :userId', { userId }).orWhere('coHost.id IS NULL');
-        })
-      )
-      .andWhere(
-        new Brackets((qb) => {
-          qb.where('participant.id != :userId', { userId }).orWhere('participant.id IS NULL');
+          qb.where('coHost.id = :userId', { userId }).orWhere('user.id = :userId', { userId });
         })
       )
       .andWhere(
         new Brackets((qb) => {
-          qb.where('host.id != :userId', { userId }).orWhere('host.id IS NULL');
+          qb.where('event.id = :eventId', { eventId });
         })
       )
-      .andWhere('event.restriction = :restriction', { restriction: String(EventRestriction.NONE) })
-      .getMany();
+      .getOne();
+    if (event) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Find all the events that the user has not joined, created or is not co-hosting
+   * @param userId
+   */
+  async findDiscoverEvent(userId: number): Promise<Event[]> {
```

Comment:
```suggestion
  async findDiscoverableEvents(userId: number): Promise<Event[]> {
```
Since you're returning an array of events (and also to match the returned variable)

Reply:
You can ignore this and we can stick [to this method name](https://github.com/RGPosadas/Mull/pull/147/files#diff-b31268315353d230075823e831673a7cc40d3c83bf1720d4b5c2b2689131e4daL54)

Reply:
Have a look at the rename I did in #147, lemme know if you don't like it or have another preference